### PR TITLE
user can add, delete product to cart in data seed

### PR DESCRIPTION
**Pull request by acv-baodq**
Demo add, delete product in Cart by ajax

### CODE REVIEW

File: app/assets/javascripts/navbar.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  })
+}
+$( document ).ready(function() {
+  $("#cart-show-btn").on('click', function(){
+    $.ajax({
+      type: "GET",
+      url: "/carts/1",
+      success: function(data, textStatus, jqXHR){
+        generate_list_product(data);
+        console.log('test',data)
```
Comment (acv-huunt) [inline app/assets/javascripts/navbar.js]: Remove console.log

File: app/models/cart.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 class Cart < ApplicationRecord
+  has_many :products
+
+  def self.get_total
+    first.products.reduce(0) { |sum, p| sum + p.price }
```
Comment (acv-huunt) [inline app/models/cart.rb]: Tai sao lai co first o cho nay.

Dung products.pluck(:price).sum

File: app/views/layouts/_cart.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div class="modal fade" id="cart" tabindex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
```
Comment (acv-huunt) [inline app/views/layouts/_cart.html.erb]: Chuyen html qua slim

File: app/models/cart.rb (hunk 2 above)
Comment (acv-baodq) [reply app/models/cart.rb]: Có first vì e đang test thôi ạ. Vì e chưa làm devise nên chưa biết cart của user nào.

File: app/assets/javascripts/navbar.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+function generate_list_product(data){
```
Comment (acv-huunt) [inline app/assets/javascripts/navbar.js]: Phan xu ly cart nen dat la cart.js

File: app/controllers/carts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    product = Product.find(params[:id].to_i)
+    @cart.products.delete(product)
+    render :json => {data: @cart.products, total: @cart.get_total, messages: "Delete #{product.name} success" }
+  end
+
+  private
+    def guest_user
+      id = session[:cart_id]
+      if id.nil?
+        @cart = Cart.create
```
Comment (acv-huunt) [inline app/controllers/carts_controller.rb]: Dung cai nay
https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by

File: app/views/products/index.html.slim [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#product-index
+  .my-paginate
+    = paginate @products, class: 'pagination'
+  .products-list
+    - @products.each do |product|
```
Comment (acv-huunt) [inline app/views/products/index.html.slim]: Dung render partial

**Pull request merged**